giturlparse/tests/parse.py

```diff
@@ -112,6 +112,15 @@
         'owner': 'Org',
         'repo': 'Repo',
 
+        'protocol': 'https',
+        'platform': 'gitlab'
+    })),
+    ('HTTPS', ('https://gitlab.com/nephila/giturlparse/blob/master/giturlparse/github.py', {
+        'host': 'gitlab.com',
+        'user': 'git',
+        'owner': 'nephila/giturlparse/blob/master/giturlparse',
+        'repo': 'github.py',
```

Comment:
If I understand correctly, previously github.py would end up in the `host` or something like that.

So while this is a bad result for `repo`, it is less bad than before this PR, and at least it is working much better for github.

IMO, "owner" shouldnt contain a `/` on GitLab either.  Subgroups should be accessed using a new member named something like `path` / `pathname`  c.f. https://github.com/retr0h/git-url-parse/pull/37   Or departing from the `urlparse` tuple, a member called `groups` which is a list of strings.

Reply:
@jayvdb this is as far as I am comfortable going with supporting the different system details

what would you think?

Reply:
Up to you.  fwiw, https://github.com/coala/git-url-parse has **just** been transferred to the @coala group, which means we can put our orgs needs into that codebase.   Which then leads to how do we cooperate/compete for the name? c.f. https://github.com/coala/git-url-parse/issues/43

Reply:
Sorry if the wording sounded a bit rude, it was not meant. It's just that regexps are a PITA to maintain beyond a certain complexity :)
I also forgot to mention that I updated the PR based on your suggestions, and this PR should be able to support more gitlab features in a more natural way.

Regarding git-url-parse, it certainly isn't super smart to have two competing and very similar implementation that covers this use case, thus definitely let's talk about this. I will answer in the linked issue to stay on topic